src/engine/Marabou.cpp

```diff
@@ -80,16 +89,27 @@ void Marabou::prepareInputQuery()
           Step 1: extract the network
         */
         String networkFilePath = Options::get()->getString( Options::INPUT_FILE_PATH );
+
         if ( !File::exists( networkFilePath ) )
         {
             printf( "Error: the specified network file (%s) doesn't exist!\n", networkFilePath.ascii() );
             throw MarabouError( MarabouError::FILE_DOESNT_EXIST, networkFilePath.ascii() );
         }
         printf( "Network: %s\n", networkFilePath.ascii() );
 
-        // For now, assume the network is given in ACAS format
-        _acasParser = new AcasParser( networkFilePath );
-        _acasParser->generateQuery( _inputQuery );
+        if ( networkFilePath.endsWith( ".onnx" ) )
```

Comment:
Maybe just cast networkFilePath to MString? You already implemented the method there and you can then get rid of the boost include, since it seems it is there solely for this call?